Reverse Transformer

rsdamm opened the pull request:
Added a transformer that will reverse the order of columns.  It also provides the nulltransformer which is simply a passthrough.  Input/output defined in the config.properties.

marinusd commented on src/main/java/com/plesba/datamanager/transformers/NullTransformer.java:
```diff
@@ -0,0 +1,72 @@
+import java.io.PipedOutputStream;
+
+public class NullTransformer {
+    private final PipedInputStream inputStream;
```
I don’t understand why one of these streams is declared ‘final’ and the other isn’t. Maybe I am missing something obvious, though.

marinusd commented on src/main/java/com/plesba/datamanager/transformers/NullTransformer.java:
```diff
@@ -0,0 +1,72 @@
+    public NullTransformer() {
+
+        LOG.info("NullTransformer started processing with no parameters");
+        inputStream = null;
```
What good would this constructor do?

marinusd commented on src/main/java/com/plesba/datamanager/transformers/NullTransformer.java:
```diff
@@ -0,0 +1,72 @@
+            LOG.info("NullTransformer reading from input stream/writing to output stream");
+
+            StringBuilder recordStringBuffer = new StringBuilder();
+            String streamRecord = new String();
```
In Java, strings are immutable, yes? So this initialization allocates a zero-length String that is never used.

marinusd commented on src/main/java/com/plesba/datamanager/transformers/ReverseTransformer.java:
```diff
@@ -0,0 +1,85 @@
+
+    public ReverseTransformer(PipedInputStream parameterInputStream, PipedOutputStream parameterOutputStream ) {
+
+        LOG.info("NullTransformer started processing");
```
Typo

marinusd commented on src/main/java/com/plesba/datamanager/transformers/ReverseTransformer.java:
```diff
@@ -0,0 +1,85 @@
+                    String[] revData = new String[arrayOfColumns.length];
+                    j=0;
+                    for (int i = arrayOfColumns.length-1; i>=0 ;i--) {
+                          revData[j]= arrayOfColumns[i];
```
Nice! Simple and clear.

rsdamm commented:
From: Marinus [mailto:notifications@github.com] 
Sent: Tuesday, August 28, 2018 4:13 AM
To: rsdamm/DataManager <DataManager@noreply.github.com>
Cc: Renee D <renee@plesba.com>; Author <author@noreply.github.com>
Subject: Re: [rsdamm/DataManager] Reverse Transformer (#13)

 

@marinusd approved this pull request.

Pleased to see all the logging cleanup. Looks like a good pattern for transformers is emerging.

  _____  

In src/main/java/com/plesba/datamanager/DataManager.java <https://github.com/rsdamm/DataManager/pull/13#discussion_r213252756> :

>  
-
-        LOG.info("DataManager datasource =  "+ datasource);
-        LOG.info("DataManager datatarget =  "+ datatarget);
+        LOG.info("DataManager datasource =  " + datasource);
+        LOG.info("DataManager datatarget =  " + datatarget);
+        LOG.info("DataManager transformtype = " + transformType);

I don’t see where the String object transformType is assigned a value. When it’s declared in line 57, the object is set null (not “null”).
What does line 83 actually output?

It comes from the properties file. Line 83 outputs what it got from the properties file.

 

  _____  

In src/main/java/com/plesba/datamanager/transformers/NullTransformer.java <https://github.com/rsdamm/DataManager/pull/13#discussion_r213254254> :

> @@ -0,0 +1,72 @@
+package com.plesba.datamanager.transformers;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+public class NullTransformer {
+    private final PipedInputStream inputStream;

I don’t understand why one of these streams is declared ‘final’ and the other isn’t. Maybe I am missing something obvious, though.

 

I just missed making them the same. 

  _____  

In src/main/java/com/plesba/datamanager/transformers/NullTransformer.java <https://github.com/rsdamm/DataManager/pull/13#discussion_r213254829> :

> +    private PipedOutputStream outputStream;
+    private int recordCount = 0;
+    private byte[] theByteArray = null;
+    private static final Log LOG = LogFactory.getLog(NullTransformer.class);
+
+    public NullTransformer(PipedInputStream parameterInputStream, PipedOutputStream parameterOutputStream ) {
+
+        LOG.info("NullTransformer started processing");
+        inputStream = parameterInputStream;
+        outputStream = parameterOutputStream;
+    }
+
+    public NullTransformer() {
+
+        LOG.info("NullTransformer started processing with no parameters");
+        inputStream = null;

What good would this constructor do?

 

I deleted it.

  _____  

In src/main/java/com/plesba/datamanager/transformers/NullTransformer.java <https://github.com/rsdamm/DataManager/pull/13#discussion_r213255732> :

> +    }
+
+    public NullTransformer() {
+
+        LOG.info("NullTransformer started processing with no parameters");
+        inputStream = null;
+        outputStream = null;
+    }
+
+    public void processDataFromInputStream() throws IOException {
+
+        try {
+            LOG.info("NullTransformer reading from input stream/writing to output stream");
+
+            StringBuilder recordStringBuffer = new StringBuilder();
+            String streamRecord = new String();

In Java, strings are immutable, yes? So this initialization allocates a zero-length String that is never used.

 

The string is used only to print something out to the log.

  _____  

In src/main/java/com/plesba/datamanager/transformers/ReverseTransformer.java <https://github.com/rsdamm/DataManager/pull/13#discussion_r213255869> :

> +
+import java.io.IOException;
+import java.io.PipedInputStream;
+import java.io.PipedOutputStream;
+
+public class ReverseTransformer {
+    private final PipedInputStream inputStream;
+    private PipedOutputStream outputStream;
+    private int recordCount = 0;
+    private byte[] theByteArray = null;
+    private static final Log LOG = LogFactory.getLog(ReverseTransformer.class);
+    private String csvDelimiter = ",";
+
+    public ReverseTransformer(PipedInputStream parameterInputStream, PipedOutputStream parameterOutputStream ) {
+
+        LOG.info("NullTransformer started processing");

Typo

 

Changed it to read ‘reverse transformer’

  _____  

In src/main/java/com/plesba/datamanager/transformers/ReverseTransformer.java <https://github.com/rsdamm/DataManager/pull/13#discussion_r213256155> :

> +            while (streamByte != -1) {
+                /* until end of stream */
+                if (streamByte != 10) {
+                    //end of line
+                    recordStringBuffer.append((char) streamByte);
+                } else {
+                    /* process record */
+                    recordCount++;
+
+                    //reverse the columns
+                    streamRecord = recordStringBuffer.toString();
+                    String[] arrayOfColumns = streamRecord.split(csvDelimiter);
+                    String[] revData = new String[arrayOfColumns.length];
+                    j=0;
+                    for (int i = arrayOfColumns.length-1; i>=0 ;i--) {
+                          revData[j]= arrayOfColumns[i];

Nice! Simple and clear.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub <https://github.com/rsdamm/DataManager/pull/13#pullrequestreview-150048539> , or mute the thread <https://github.com/notifications/unsubscribe-auth/APsYaES8PWQwSd-wZNEHvdBcW7ZCW-Yyks5uVReggaJpZM4WM_zv> .
